From 73e30222bdce9ac66e16c0e2b524856f2cf5aa38 Mon Sep 17 00:00:00 2001
From: Alexander von Gluck IV <kallisti5@unixzen.com>
Date: Sat, 15 Apr 2023 08:08:43 -0500
Subject: [PATCH] gallium/frontends/hgl: Convert to system GLView.h.

* This is needed for our BGL_INDEX, BGL_ALPHA, BGL_DEPTH, etc.
---
 meson.build                           | 1 +
 src/gallium/frontends/hgl/hgl.c       | 3 +--
 src/gallium/frontends/hgl/meson.build | 1 +
 3 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index c7a4be82e59..7610defb896 100644
--- a/meson.build
+++ b/meson.build
@@ -1524,6 +1524,7 @@ endif
 
 if with_platform_haiku
   dep_network = cc.find_library('network')
+  dep_glkit = dependency('gl', version : '>= 1.0')
 endif
 
 dep_futex = null_dep
diff --git a/src/gallium/frontends/hgl/hgl.c b/src/gallium/frontends/hgl/hgl.c
index 7fd5745d90f..a730593ba04 100644
--- a/src/gallium/frontends/hgl/hgl.c
+++ b/src/gallium/frontends/hgl/hgl.c
@@ -10,6 +10,7 @@
 #include "hgl_context.h"
 
 #include <stdio.h>
+#include <GLView.h>
 
 #include "util/format/u_formats.h"
 #include "util/u_atomic.h"
@@ -18,8 +19,6 @@
 #include "util/u_inlines.h"
 #include "state_tracker/st_context.h"
 
-#include "GLView.h"
-
 
 #ifdef DEBUG
 #   define TRACE(x...) printf("hgl:frontend: " x)
diff --git a/src/gallium/frontends/hgl/meson.build b/src/gallium/frontends/hgl/meson.build
index 3b6348db338..ac4540558c4 100644
--- a/src/gallium/frontends/hgl/meson.build
+++ b/src/gallium/frontends/hgl/meson.build
@@ -31,6 +31,7 @@ endif
 libsthgl = static_library(
   'sthgl',
   files_libsthgl,
+  dependencies : [dep_glkit],
   include_directories : [
     inc_include, inc_util, inc_mesa, inc_mapi, inc_src,
     inc_gallium, inc_gallium_aux
-- 
2.37.3

